genie-web/src/main/java/com/netflix/genie/web/spring/autoconfigure/scripts/ScriptsAutoConfiguration.java

```diff
@@ -89,14 +90,12 @@ ScriptManager scriptManager(
      * Create a {@link SmartInitializingSingleton} that "warms" known scripts so they're ready for execution on first
      * invocation.
      *
-     * @param managedScrips the managed scripts, if any exist in context
+     * @param managedScripts the managed scripts, if any exist in context
      * @return A {@link ManagedScriptPreLoader} that runs after the application context is ready
      */
     @Bean
-    public ManagedScriptPreLoader managedScriptPreLoader(
-        final Optional<List<ManagedScript>> managedScrips
-    ) {
-        return new ManagedScriptPreLoader(managedScrips.orElse(Lists.newArrayList()));
+    public ManagedScriptPreLoader managedScriptPreLoader(final List<ManagedScript> managedScripts) {
```

Comment:
This will make application context initialization fail if genie is not configured to use at least one script.

Reply:
Why do you say that? The tests for the auto configuration test that no beans of type `ManagedScript` exist and yet it comes up fine. So do the smoke tests for `genie-app`.

In the case of injection of collections if spring doesn't have any beans matching the requested type. In this case `ManagedScript` it will just provide an empty list. The logic in this preload bean is just to loop through the list and call `warmUp`. So it all becomes a no-op.

Reply:
It's something i bumped against, tho it's possible it has changed recently.


https://www.baeldung.com/spring-injecting-collections
 > Now, let's consider a scenario when there is not a BaeldungBean. If there isn't a BaeldungBean registered in the application context, Spring will throw an exception because the required dependency is missing.

Also: https://stackoverflow.com/questions/19299114/is-there-an-easy-way-to-autowire-empty-collection-if-no-beans-present-in-spring

Admittedly, i cannot explain how those tests can pass..

Reply:
Ah i think tests pass because, `genie-web-defaults.yml` actually does declare scripts:

```
  scripts:
    cluster-selector:
      source:
      auto-load-enabled: false
      timeout: 5000
    execution-mode-filter:
      source:
      auto-load-enabled: false
      timeout: 5000
```

They have an empty `source` field so they'll throw if not invoked, but the beans do get created.

Reply:
The genie-web-defaults should not be considered by the `AutoConfiguration` tests as they're not called out as property sources and the post config `processors` that loads them is not activated. The test explicitly asserts that there are no beans of those types:
https://github.com/Netflix/genie/blob/master/genie-web/src/test/java/com/netflix/genie/web/spring/autoconfigure/scripts/ScriptsAutoConfigurationTest.java#L65

We should take those blank strings out of the defaults. I'm not sure why we would put them in there like that. `matchIfMissing` is false by default so if they're gone they won't match.

Reply:
The properties explain why `genie-app` test works.
My guess as to why `*AutoConfiguration` work was that none actually loads that configuration in an environment with no scripts.

> I'm not sure why we would put them in there like that.
Convenience of injection. I know it's not pretty to have those empty properties by default, but the current scheme allows to:
 - Set a property for a script and have everything work
 - Provide a custom *ManagedScript bean to override the default
 - Provide a custom *Selector to replace the one based on script

i.e. it allows override/inject at all 3 levels.
That empty default property is the price to pay in terms of ugliness.. I thought it was a fair price.